tempesta_fw/work_queue.h

```diff
@@ -77,6 +86,11 @@ tfw_wq_push(TfwRBQueue *q, void *ptr, int cpu, struct irq_work *work,
 	long ticket = __tfw_wq_push(q, ptr);
 	if (unlikely(ticket))
 		return ticket;
+	/*
+	 * The atomic operation is 'atomic64_cmpxchg()' in
+	 * '__tfw_wq_push()' above.
```

Comment:
Why? `atomic64_cmpxchg()` is called very rarely, only on slow path. In fast path there are just atomic reads and writes: in good case there is write barrier, but in bad case there is no any barriers.

Reply:
Fast path take place only when the queue is full and `tfw_wq_push` exits with ticket - before IPI generation. In case of insertion new element into the queue - the `atomic64_cmpxchg()` must be called.